pkg/controllers/user/rbac/namespace_handler.go

```diff
@@ -118,6 +120,39 @@ func (n *nsLifecycle) ensurePRTBAddToNamespace(ns *v1.Namespace) (bool, error) {
 			return false, errors.Wrapf(err, "couldn't ensure binding %v in %v", prtb.Name, ns.Name)
 		}
 	}
+
+	var namespace string
+	if parts := strings.SplitN(projectID, ":", 2); len(parts) == 2 && len(parts[1]) > 0 {
+		namespace = parts[1]
+	} else {
+		return hasPRTBs, nil
+	}
+
+	rbs, err := n.m.rbLister.List(ns.Name, labels.Set(map[string]string{}).AsSelector())
+	if err != nil {
+		return false, errors.Wrapf(err, "couldn't list role bindings in %s", ns.Name)
+	}
+	client := n.m.workload.RBAC.RoleBindings(ns.Name).ObjectClient()
+
+	for _, rb := range rbs {
+		if uid := convert.ToString(rb.Labels[rtbOwnerAnn]); uid != "" {
+			prtbs, err := n.m.prtbIndexer.ByIndex(prtbByUIDIndex, uid)
+			if err != nil {
```

Comment:
Only continue if err != nil and its a not found error (apierrors.IsNotFound(err)) if it is any other type of error, actually error out and return the error (unless you have some other specific error scenario you are concerned with here).